src/Runner.Listener/Runner.cs

```diff
@@ -477,7 +478,35 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     // todo: add retries https://github.com/github/actions-broker/issues/49
                                     var runServer = HostContext.CreateService<IRunServer>();
                                     await runServer.ConnectAsync(new Uri(settings.ServerUrl), creds);
-                                    var jobMessage = await runServer.GetJobMessageAsync(messageRef.RunnerRequestId);
+                                    Pipelines.AgentJobRequestMessage jobMessage = null;
+
+                                    var remainingRetryLimitTime = TimeSpan.FromMinutes(5);
+                                    while (true)
+                                    {
+                                        try
+                                        {
+                                            jobMessage = await runServer.GetJobMessageAsync(messageRef.RunnerRequestId);
```

Comment:
Info: What happens now in case `runServer.GetJobMessageAsync` throws?
Doesn't the loop at `while (!HostContext.RunnerShutdownToken.IsCancellationRequested)` at Line 337 handle retries?

Reply:
The loop on 337 always tries to retrieve latest message from MessageListener.
It is hard and is not worth to make it not retrieve latest message if we want to retry only retrieving message from broker.

> Info: What happens now in case `runServer.GetJobMessageAsync` throws?

Do you mean before the change? Right now it will be skipped. Current message from messagelistener will be deleted, and in the new iteration we will work already with other message